pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -244,6 +244,9 @@ func syncNamespace(kubeClient clientset.Interface, versions *unversioned.APIVers
 	// we have removed content, so mark it finalized by us
 	result, err := retryOnConflictError(kubeClient, namespace, finalizeNamespaceFunc)
 	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
I'm not convinved about this - if we return "not found" error, it doesn't meen that it finished succefully - there might still be objects to remove. Or am I missing something?